era_5g_interface/task_handler_internal_q.py

```diff
@@ -45,13 +67,40 @@ def store_image(self, metadata: dict, image: np.ndarray) -> None:
                 The format is NetApp-specific.
             image (_type_): The image to be processed.
         """
+
+        # make sure multiple calls of store_image do not interfere with each other
+        self._q_put_lock.acquire()
```

Comment:
Why not `with self._q_put_lock:`?

Reply:
Yes, I agree, it makes the code cleaner. Changed.